modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TestTcpDiscoverySpi.java

```diff
@@ -31,6 +38,9 @@
     /** */
     public boolean ignorePingResponse;
 
+    /** Interceptors of discovery messages. */
+    private List<DiscoveryHook> discoHooks = new ArrayList<>();
```

Comment:
Let's allocate discoHooks on the first hook addition.

Reply:
Thanks. Done.